lib/board_renderer.rb

```diff
@@ -0,0 +1,72 @@
+require_relative './board.rb'
+require_relative './renderer.rb'
+
+class BoardRenderer < Renderer
+    def render(board)
+        pre_result_storage = []
+        line_storage = []
+
+        for i in 0..(board.data.length - 1)
+            if ((_adjust_index_to_position(i)%board.width) == 0)
+                fill_string = _make_space_buffer(4) + board.data[i] + _make_space_buffer(4)
+                line_storage.push(fill_string + "\n")
+                pre_result_storage.push(_create_n_nonline_rows(board, 1) + line_storage.join("|") + _create_n_nonline_rows(board, 1))
+                line_storage = []
+            else
+                fill_string = _make_space_buffer(4) + board.data[i] + _make_space_buffer(4)
+                line_storage.push(fill_string)
+            end
+        end
+
+        return pre_result_storage.join(_create_line_row(board))
+    end
+
+    def _make_space_buffer(count)
+        buffer = ""
+        for counter in 1..count
+            buffer += " "
+        end
+        return buffer
+    end
+
+    def _create_nonline_row(board)
+        row_line = ""
+        for i in 0..(board.width-1)
+            if i < board.width - 1
+                row_line += _make_space_buffer(9) + "|"
+            else
+                row_line += _make_space_buffer(9) + "\n"
+            end
+        end
+        return row_line
+    end
+
+    def _create_n_nonline_rows(board, count)
+        fill = ""
+        for i in 0..(count - 1)
```

Comment:
I think some form of map and join could be cleaner here

Reply:
Yeah, ruby [`Enumerable`](https://ruby-doc.org/core-2.2.0/Enumerable.html) methods can offer some very succinct solutions.

This might be a good place to point out that [scoping is weird](https://github.com/rubocop-hq/ruby-style-guide#no-for-loops) in ruby **`for`** loops.